app/libraries/AdminUtilities.php

```diff
@@ -58,39 +97,80 @@ public static function get_gateway_approval_statuses()
         return $gatewayApprovalStatusObject::$__names;
     }
 
+    public static function update_form( $gatewayId, $gatewayData){
+
+        if( isset( $gatewayData["updateRequest"])) {
+            return TenantProfileService::getGateway( Session::get('authz-token'), $gatewayId);
+        }
+
+    }
+
+    public static function user_update_gateway( $gatewayId, $gatewayData){
+        $gateway = TenantProfileService::getGateway( Session::get('authz-token'), $gatewayId);
+        $gateway->gatewayApprovalStatus = GatewayApprovalStatus::APPROVED;
+        $gateway->emailAddress = $gatewayData["email-address"];
+        $gateway->gatewayURL = $gatewayData["gateway-url"];
+        $gateway->identityServerUserName = $gatewayData["admin-username"];
+        $gateway->identityServerPasswordToken  = $gatewayData["admin-password"];
+        $gateway->gatewayAdminFirstName = $gatewayData["admin-firstname"];
+        $gateway->gatewayAdminLastName = $gatewayData["admin-lastname"];
+        $gateway->gatewayAdminEmail = $gatewayData["admin-email"];
+        $gateway->reviewProposalDescription = $gatewayData["project-details"];
+        $gateway->gatewayPublicAbstract = $gatewayData["public-project-description"];
+        if( TenantProfileService::updateGateway( Session::get('authz-token'), $gateway) ){
+            return 1;
+        }
+        else{
+            //Need to find a better way for this.
+            // retun echo "Tenant Name is already in use";
+            return -1;
+        }
+    }
+
     public static function update_gateway( $gatewayId, $gatewayData){
 
-        $gateway = Airavata::getGateway( Session::get('authz-token'), $gatewayId);
+        $gateway = TenantProfileService::getGateway( Session::get('authz-token'), $gatewayId);
```

Comment:
I don't think this will work to get a gateway by gatewayId and then update it, when it is possible for multiple gateways to have the same id.  I think we need a backend method to get a gateway by the internal id.  What do you think? I can add it.